### PR TITLE
New plugin: Active Players

### PR DESCRIPTION
**Pull request by davidryan59**
Plugin uses GraphQL to query a list of the 1000 latest arrivals (moves)

A table is created of which players have sent the most moves, in decreasing frequency

Useful for:
- Telling who is most dangerous in the heat of battle (i.e. capacity to send moves)
- After the round is over, being able to find out who is still active in the arena

### CODE REVIEW

**Comment by ichub** [comment]
Hey @davidryan59 - we released a broken subgraph for round 5, but it's since been fixed! Do you think you could test your plugin out with the following subgraph? https://thegraph.com/hosted-service/subgraph/darkforest-eth/dark-forest-v06-round-5

File: content/utilities/active-players/plugin.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+const PLUGIN_NAME = "Active Players";
+const DEV_MODE = false;  // Put as true to highlight UI sections for debugging
+const ADDRESS_CHARS_TO_DISPLAY = 6;
+const OWNER_CHARS_TO_DISPLAY = 18;
+const MAX_RECORD_COUNT = 1000;  // GraphQL will retrieve up to 1000 objects in one go
+
+// -----------------------------
+
+
+const pg = df.getProcgenUtils();
```
Comment (snowtigersoft) [inline content/utilities/active-players/plugin.js:23]: this function is removed in v0.6.5

Comment (snowtigersoft) [inline content/utilities/active-players/plugin.js:23]: this function is removed in v0.6.5

File: content/utilities/active-players/plugin.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    searchRow.appendChild(searchButton);
+    container.appendChild(searchRow);
+    container.appendChild(this.tableRows);
+  }
+
+  destroy() {
+  }
+}
+
+export default ActivePlayers;
```
Comment (cristobal) [inline content/utilities/active-players/plugin.js:144]:
```suggestion
export default ActivePlayers;

```
nitpick github nags about newline.
Screenshot 2022-03-31 at 14 17 39

File: content/utilities/active-players/plugin.js (hunk 1 above)
Comment (cristobal) [reply content/utilities/active-players/plugin.js:23]: @davidryan59 added a PR on your branch that fixes the `pg.getProcgenUtils()` part. https://github.com/davidryan59/plugins/pull/2

Comment (davidryan59) [reply content/utilities/active-players/plugin.js]: I've merged @cristobal's PR, this should be fixed now

**Comment by davidryan59** [comment]
> Hey @davidryan59 - we released a broken subgraph for round 5, but it's since been fixed! Do you think you could test your plugin out with the following subgraph? https://thegraph.com/hosted-service/subgraph/darkforest-eth/dark-forest-v06-round-5

@ichub I have tried using the graph API above, and it did not work, with error:

```
Access to fetch at 'https://thegraph.com/hosted-service/subgraph/darkforest-eth/dark-forest-v06-round-5' from origin 'https://zkga.me' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

It worked with this Graph API: `https://api.thegraph.com/subgraphs/name/darkforest-eth/dark-forest-v06-round-5`

I think this line will need to update in any future rounds anyway.

How would you like to proceed?

**Comment by davidryan59** [comment]
From @ichub via Discord: "We could add `df.getGraphApiUrl()`. I can get that in tomorrow"

**Comment by ichub** [comment]
> From @ichub via Discord: "We could add `df.getGraphApiUrl()`. I can get that in tomorrow"

done! we've added the following API:

```ts
import {THEGRAPH_API_URL} from "https://cdn.skypack.dev/@darkforest_eth/constants";
console.log(THEGRAPH_API_URL);
```

**Comment by davidryan59** [comment]
@ichub I have used THEGRAPH_API_URL import now, tested in v0.6.5, it works well. Thanks!

Ready for review now?

**Review by ichub — APPROVED** [review]

**Pull request merged**